tensorflow_federated/python/core/utils/execution_tracing.py

```diff
@@ -0,0 +1,313 @@
+import asyncio
+import contextlib
+from enum import Enum
+import functools
+import inspect
+import logging
+import re
+
+from tensorflow.python.framework.ops import EagerTensor as tf_EagerTensor
+
+from tensorflow_federated.proto.v0 import computation_pb2
+from tensorflow_federated.python.common_libs import anonymous_tuple
+from tensorflow_federated.python.common_libs import tracing
+from tensorflow_federated.python.core.api import computation_types
+from tensorflow_federated.python.core.impl import computation_impl
+from tensorflow_federated.python.core.impl.compiler import intrinsic_defs
+from tensorflow_federated.python.core.impl.executors import caching_executor
+from tensorflow_federated.python.core.impl.executors import composing_executor
+from tensorflow_federated.python.core.impl.executors import eager_tf_executor
+from tensorflow_federated.python.core.impl.executors import federating_executor
+from tensorflow_federated.python.core.impl.executors import reference_resolving_executor
+from tensorflow_federated.python.core.impl.executors import remote_executor
+from tensorflow_federated.python.core.impl.executors import sizing_executor
+from tensorflow_federated.python.core.impl.executors import thread_delegating_executor
+from tensorflow_federated.python.core.impl.executors import transforming_executor
+
+
+class TraceFilterMode(Enum):
+  VALUES = 0
+  INTRINSICS = 1
+  ALL = 2
+
+
+def pattern_union(*patterns):
+  pattern_strs = map(lambda p: p.pattern, patterns)
+  return re.compile("|".join(pattern_strs))
```

Comment:
Can you use single quotes throughout the change consistently? I'm not sure that we have 100% parity between internal and external lint, but I know this fails internally.